src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -157,18 +154,48 @@ private void VerifySigning(string nupkgPath)
                 return;
             }
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!_isNuGetTool)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
+                        out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
+                                                                 Environment.NewLine +
+                                                                 commandOutput);
+                    }
+                }
+            }
+            else
             {
-                if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
-                    out string commandOutput))
+                if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Given that NuGet package signature verification is enabled by default on Linux in .NET 8 https://github.com/dotnet/sdk/pull/31868 PR, I think we could enable signature verification for .NET Tools NuGet packages on Linux also.

cc @dtivel

Reply:
Can the default behavior be the same as https://github.com/dotnet/sdk/pull/31868/files, driven by the same environment variable?